app/screens/send-bitcoin-screen/use-send-payment.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  PaymentSendResult,
+  useIntraLedgerPaymentSendMutation,
+  useIntraLedgerUsdPaymentSendMutation,
+  useLnInvoicePaymentSendMutation,
+  useLnNoAmountInvoicePaymentSendMutation,
+  useLnNoAmountUsdInvoicePaymentSendMutation,
+  useOnChainPaymentSendMutation,
+} from "@app/graphql/generated"
+import { useState, useMemo } from "react"
+import { SendPayment } from "./payment-details/index.types"
+
+export type UseSendPaymentResult = {
+  loading: boolean
+  sendPayment:
+    | (() => Promise<{
+        status: PaymentSendResult | null | undefined
+        errorsMessage?: string
+      }>)
+    | undefined
+    | null
+}
+
+export const useSendPayment = (
+  sendPaymentFn?: SendPayment | null,
+): UseSendPaymentResult => {
+  const [intraLedgerPaymentSend] = useIntraLedgerPaymentSendMutation()
+  const [intraLedgerUsdPaymentSend] = useIntraLedgerUsdPaymentSendMutation()
+  const [lnInvoicePaymentSend] = useLnInvoicePaymentSendMutation()
+  const [lnNoAmountInvoicePaymentSend] = useLnNoAmountInvoicePaymentSendMutation()
+  const [lnNoAmountUsdInvoicePaymentSend] = useLnNoAmountUsdInvoicePaymentSendMutation()
+  const [onChainPaymentSend] = useOnChainPaymentSendMutation()
+  const [loading, setLoading] = useState(false)
+
+  const sendPayment = useMemo(() => {
+    return (
+      sendPaymentFn &&
+      (async () => {
+        setLoading(true)
+        const response = await sendPaymentFn({
```

Comment:
doesn't `sendPaymentFn` already have `loading`?  this might be better than managing the state explicitly. ie if throw for whatever reason, I guess loading would be correctly update if fetch from mutation, but not as it's currently implemented.

Reply:
> some comments skimming through the PR. I didn't really dive into it.
> 
> we haven't made any or much test on hooks until now, but maybe this could be a good place to start?

Yeah if the team feels that this is a good approach for the refactor, it will be really easy to unit test,  and I plan on writing them.

Reply:
Switched to using the `loading` from the mutation